test/e2e/flowexporter_test.go

```diff
@@ -0,0 +1,166 @@
+	DATA RECORD:
+	 template id:  256
+	 nfields:      21
+	 sourceIPv4Address: 100.10.0.117
+	 destinationIPv4Address: 100.10.1.128
+	 sourceTransportPort: 44586
+	 destinationTransportPort: 8080
+	 protocolIdentifier: 6
+	 packetTotalCount: 7
+	 octetTotalCount: 420
+	 packetDeltaCount: 0
+	 octetDeltaCount: 0
+	 55829_101: 0x7765622d636c69
+	IPFIX-HDR:
+	 version=10, length=119
+	 unixtime=1596608558 (2020-08-04 23:22:38 PDT)
+	 seqno=159, odid=1269807227
+	DATA RECORD:
+	 template id:  256
+	 nfields:      21
+	 sourceIPv4Address: 100.10.0.114
+	 destinationIPv4Address: 100.10.1.127
+	 sourceTransportPort: 42872
+	 destinationTransportPort: 8080
+	 protocolIdentifier: 6
+	 packetTotalCount: 7
+	 octetTotalCount: 420
+	 packetDeltaCount: 0
+	 octetDeltaCount: 0
+	*/
+	re := regexp.MustCompile("(?m)^.*" + "#" + ".*$[\r\n]+")
+	collectorOutput = re.ReplaceAllString(collectorOutput, "")
+	collectorOutput = strings.TrimSpace(collectorOutput)
+	recordSlices := strings.Split(collectorOutput, "IPFIX-HDR:")
+	// Delete the first element from recordSlices
+	recordSlices[0] = recordSlices[len(recordSlices)-1]
+	recordSlices[len(recordSlices)-1] = ""
+	recordSlices = recordSlices[:len(recordSlices)-1]
+	// Iterate over recordSlices and build some results to test with expected results
+	templateRecords := 0
+	dataRecordsIntraNode := 0
+	for _, record := range recordSlices {
+		if strings.Contains(record, "TEMPLATE RECORD") {
+			templateRecords = templateRecords + 1
+		}
+
+		if strings.Contains(record, podAIP) && strings.Contains(record, podBIP) {
+			dataRecordsIntraNode = dataRecordsIntraNode + 1
+			// Check if records have both Pod name and Pod namespace or not
+			if !strings.Contains(record, hex.EncodeToString([]byte("perftest-a"))) {
+				t.Fatalf("Records with podAIP does not have pod name")
+			}
+			if !strings.Contains(record, hex.EncodeToString([]byte("perftest-b"))) {
+				t.Fatalf("Records with podBIP does not have pod name")
+			}
+			if !strings.Contains(record, hex.EncodeToString([]byte(testNamespace))) {
+				t.Fatalf("Records with podAIP and podBIP does not have pod namespace")
+			}
+			// Check the bandwidth using octetDeltaCount in data records sent in second ipfix interval
+			if strings.Contains(record, "seqno=2") || strings.Contains(record, "seqno=3") {
+				// One of them has no bytes ignore that
+				if !strings.Contains(record, "octetDeltaCount: 0") {
+					//split the record in lines to compute bandwidth
+					splitLines := strings.Split(record, "\n")
+					for _, line := range splitLines {
+						if strings.Contains(line, "octetDeltaCount") {
+							lineSlice := strings.Split(line, ":")
+							deltaBytes, err := strconv.ParseFloat(strings.TrimSpace(lineSlice[1]), 64)
+							if err != nil {
+								t.Fatalf("Error in converting octetDeltaCount to int type")
+							}
+							// compute the bandwidth using 5s as interval
+							recBandwidth := (deltaBytes * 8.0) / float64((int64(5.0))*time.Second.Nanoseconds())
```

Comment:
```
recBandwidth := (deltaBytes * 8.0) / float64(5*time.Second.Nanoseconds())
```
does not work?

Reply:
Yes it works. For some reason, I started with 5.0 to make everything float and navigated around that. Changed.

Reply:
Missed this change. As tests have finished in this PR, modified this file in this PR #984 .